src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -963,4 +987,37 @@ bool EventPipe::IsLockOwnedByCurrentThread()
 }
 #endif
 
+// This method will block runtime bring-up IFF DOTNET_DiagnosticsMonitorAddress != nullptr and DOTNET_DiagnosticsMonitorStopOnStart!=0 (it's default state)
+// The s_ResumeRuntimeStartupEvent event will be signaled when the Diagnostics Monitor uses the ResumeRuntime Diagnostics IPC Command
+void EventPipe::PauseForTracingAgent()
+{
+    LIMITED_METHOD_CONTRACT;
```

Comment:
LIMITED_METHOD_CONTRACT implies any methods it calls are also limited and that it is legal to call in any GC mode (those things aren't true here). Check out https://github.com/dotnet/runtime/blob/master/src/coreclr/src/inc/contract.h#L124

```suggestion
    CONTRACTL 
    {
      NOTHROW;
      GC_NOTRIGGER;
      MODE_PREEMPTIVE;
    } 
    CONTRACTL_END;
```